### PR TITLE
Firmware rn d

### PR DESCRIPTION
**Pull request by NsubugaAnold**
Hello Team, 

I worked on having bidirectional communication using the thingspeak mqtt broker: Git didnot allow having my broker credentials so for testing, you might have to use your own details. The testing was made using esp32's. Per this code. I am actually using a single esp32 as a subscriber and publisher at the same time.  You can replicate this same project on arduino by changing from Hardware serial to software Serial

### CODE REVIEW

**Comment by Baalmart** [comment]
Thanks @NsubugaAnold for this development. As for testing, we are not supposed to share credentials in the codebase since this project is opensource so hackers can easily misuse your information and compromise our entire network in the process.

File: RnD firmware/mqtt_thingspeak_bi_directional_communication/mqtt_thingspeak_bi_directional_communication.ino [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+#define TINY_GSM_MODEM_SIM800
+
+#include <PubSubClient.h> 
+#include <TinyGsmClient.h> 
+#include <ArduinoJson.h> 
+
+#define Serialgsm Serial1
+const char apn[] = "internet"; // apn for uganda 
+const char* broker = "mqtt3.thingspeak.com"; 
+const char clientID[] = "";
```
Comment (deogratius2phantom) [inline RnD firmware/mqtt_thingspeak_bi_directional_communication/mqtt_thingspeak_bi_directional_communication.ino:11]: smart to have left  out the credentials in the PR

**Review by deogratius2phantom — APPROVED** [review]
mqtt can be adopted for our network managment

**Comment by deogratius2phantom** [comment]
going ahead to merge this into staging

**Pull request merged**